### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by DmytroBondariev**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1 +1,43 @@
...
+            print(f"Eating {self.appetite} food points...")
+            return self.appetite
+        return 0
+
+
+class Cat(Animal):
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, 3, is_hungry)
+
+    def catch_mouse(self) -> None:
```
Comment (katevasilenko) [inline app/main.py:27]: This method is staticmethod

File: app/main.py [hunk 2]
```diff
@@ -1 +1,43 @@
...
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, 7, is_hungry)
+
+    def bring_slippers(self) -> None:
```
Comment (katevasilenko) [inline app/main.py:35]: This method is staticmethod

File: app/main.py [hunk 3]
```diff
@@ -1 +1,43 @@
...
+        print("The slippers delivered!")
+
+
+def feed_animals(list_of_animals: list[Animal]) -> int:
+    food_points = 0
+    for animal in list_of_animals:
+        food_points += int(animal.feed())
+    return food_points
```
Comment (katevasilenko) [inline app/main.py:43]: Use comprehension with sum() function here

**Review by katevasilenko — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 4]
```diff
@@ -24,20 +24,20 @@ class Cat(Animal):
...
         print("The slippers delivered!")
 
 
 def feed_animals(list_of_animals: list[Animal]) -> int:
-    food_points = 0
-    for animal in list_of_animals:
-        food_points += int(animal.feed())
-    return food_points
+    food_points_list = [int(animal.feed()) for animal in list_of_animals]
+    return sum(food_points_list)
```
Comment (vasylhnatiuk) [inline app/main.py:43]: you can do it to use only sum: `return sum(int(animal.feed()) for animal in list_of_animals)
 `

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 5]
```diff
@@ -1 +1,42 @@
...
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, 7, is_hungry)
+
+    @staticmethod
+    def bring_slippers() -> None:
+        print("The slippers delivered!")
+
+
+def feed_animals(list_of_animals: list[Animal]) -> int:
+    return sum(int(animal.feed()) for animal in list_of_animals)
```
Comment (Vasyl-Smutok) [inline app/main.py:42]:
```suggestion
    return sum(animal.feed() for animal in list_of_animals)
```

**Review by Vasyl-Smutok — CHANGES_REQUESTED** [review]

**Review by Vasyl-Smutok — APPROVED** [review]
Good Job 👍

**Pull request closed without merging**